CRM/SMS/BAO/Provider.php

```diff
@@ -92,6 +93,9 @@ public static function getProviders($selectArr = NULL, $filter = NULL, $getActiv
   public static function saveRecord($values) {
     $dao = new CRM_SMS_DAO_Provider();
     $dao->copyValues($values);
+    if (empty($dao->domain_id)) {
+      $dao->domain_id = CRM_Core_Config::domainID();
```

Comment:
Yes but here your saving a record into the db so i guess the question is should we update NULL values = the current domain or leave them as they are?